service/history/workflow/update/registry.go

```diff
@@ -100,7 +103,7 @@ type (
 	registry struct {
 		mu               sync.RWMutex
 		updates          map[string]*Update
-		getStoreFn       func() Store
+		getStoreFn       func() Store // TODO: revert it back to Store
```

Comment:
It used to be just `Store` here, not `func`. Then @yycptt changed it to getter func because we faced stale MS issue. Now, when registry is guaranteed to use latest MS this line can be changed back to:
```suggestion
		store       Store
```
I just don't want to do it in this PR and will do as a follow up item.